tgui/packages/tgui/interfaces/RequestManager.tsx

```diff
@@ -3,37 +3,66 @@
  * @copyright 2021 bobbahbrown (https://github.com/bobbahbrown)
  * @license MIT
  */
-import { decodeHtmlEntities } from 'common/string';
+import { createSearch, decodeHtmlEntities } from 'common/string';
 import { useState } from 'react';
 
-import { useBackend, useLocalState } from '../backend';
-import { Button, Input, Popper, Section, Table } from '../components';
+import { useBackend } from '../backend';
+import { Button, Input, Popper, Section, Stack, Table } from '../components';
 import { Window } from '../layouts';
 
+type Data = {
+  requests: Request[];
+};
+
+type Request = {
+  id: string;
+  req_type: string;
+  owner: string;
+  owner_ckey: string;
+  owner_name: string;
+  message: string;
+  additional_info: string;
+  timestamp: number;
+  timestamp_str: string;
+};
+
+const displayTypeMap = {
+  request_prayer: 'PRAYER',
+  request_centcom: 'CENTCOM',
+  request_syndicate: 'SYNDICATE',
+  request_nuke: 'NUKE CODE',
+  request_fax: 'FAX',
+  request_internet_sound: 'INTERNET SOUND',
+};
+
 export const RequestManager = (props) => {
-  const { act, data } = useBackend();
-  const { requests } = data;
-  const [filteredTypes, _] = useLocalState(
-    'filteredTypes',
+  const { act, data } = useBackend<Data>();
+  const { requests = [] } = data;
+  const [filteredTypes, setFilteredTypes] = useState(
     Object.fromEntries(
       Object.entries(displayTypeMap).map(([type, _]) => [type, true]),
     ),
   );
   const [searchText, setSearchText] = useState('');
 
+  const updateFilter = (type) => {
+    const newFilter = { ...filteredTypes };
+    newFilter[type] = !newFilter[type];
+    setFilteredTypes(newFilter);
+  };
+
   // Handle filtering
   let displayedRequests = requests.filter(
     (request) => filteredTypes[request.req_type],
   );
-  if (searchText) {
-    const filterText = searchText.toLowerCase();
-    displayedRequests = displayedRequests.filter(
-      (request) =>
-        decodeHtmlEntities(request.message)
-          .toLowerCase()
-          .includes(filterText) ||
-        request.owner_name.toLowerCase().includes(filterText),
-    );
+  const search = createSearch(
+    searchText.toLowerCase(),
```

Comment:
toLowerCase() is implied with createSearch, same with the stringifier. should be equal without
```suggestion
    searchText,
```